examples/ransomware_detection/stages/create_features.py

```diff
@@ -82,88 +82,37 @@ def accepted_types(self) -> typing.Tuple:
 
-    def on_next(self, x: AppShieldMessageMeta):
 
-        snapshot_fea_dfs = []
+    def on_next(self, x: AppShieldMessageMeta):
 
         df = x.df
 
-        # Type cast CommitCharge.
-        df["CommitCharge"] = df["CommitCharge"].astype("float").astype("Int32")
         df["Name"] = df["Name"].str.lower()
 
         # Create PID_Process feature.
-        df['PID_Process'] = df.PID + '_' + df.Process
-
-        snapshot_ids = df.snapshot_id.unique()
-
-        if len(snapshot_ids) > 1:
-            # Group snapshot rows using snapshot id.
-            all_dfs = [df[df.snapshot_id == snapshot_id] for snapshot_id in snapshot_ids]
-        else:
-            all_dfs = [df]
-
-        extract_func = self._fe.extract_features
-        combine_func = FeatureExtractor.combine_features
+        df['PID_Process'] = df.PID.astype(str) + '_'# + df.Process
 
-        # Schedule dask task `extract_features` per snapshot.
-        snapshot_fea_dfs = self._client.map(extract_func, all_dfs, feas_all_zeros=self._feas_all_zeros)
-
-        # Combined `extract_features` results.
-        features_df = self._client.submit(combine_func, snapshot_fea_dfs)
-
-        # Gather features from all the snapshots.
-        features_df = features_df.result()
-
-        # Snapshot sequence will be generated using `source_pid_process`.
-        # Determines which source generated the snapshot messages.
-        # There's a chance of receiving the same snapshots names from multiple sources(hosts)
+        features_df = self._fe.extract_features(df, feas_all_zeros=self._feas_all_zeros)
         features_df['source_pid_process'] = x.source + '_' + features_df.pid_process
-
-        # Sort entries by pid_process and snapshot_id
-        features_df = features_df.sort_values(by=["pid_process", "snapshot_id"]).reset_index(drop=True)
+        features_df.index = features_df.source_pid_process
 
         # Create AppShieldMessageMeta with extracted features information.
         meta = AppShieldMessageMeta(features_df, x.source)
 
         return meta
 
-    def create_multi_messages(self, x: AppShieldMessageMeta) -> typing.List[MultiMessage]:
-
-        multi_messages = []
-
-        df = x.df
-
-        pid_processes = df.pid_process.unique()
-
-        # Create multi messaage per pid_process, this assumes that the DF has been sorted by the `pid_process` column
-        for pid_process in pid_processes:
-
-            pid_process_index = df[df.pid_process == pid_process].index
-
-            start = pid_process_index.min()
-            stop = pid_process_index.max() + 1
-            mess_count = stop - start
-
-            multi_message = MultiMessage(meta=x, mess_offset=start, mess_count=mess_count)
-            multi_messages.append(multi_message)
-
-        return multi_messages
-
-    def on_completed(self):
+    def on_completed():
         # Close dask client when pipeline initiates shutdown
-        self._client.close()
+        pass
+        # self._client.close()
```

Comment:
More excess comments

Reply:
Can you please remove the on_completed function as we are not using Dask for the `create_features` stage?